src/Admin/SiteHealth.php

```diff
@@ -49,15 +49,42 @@ public function add_tests( $tests ) {
 			'label' => esc_html__( 'cURL multi functions', 'amp' ),
 			'test'  => [ $this, 'curl_multi_functions' ],
 		];
-		$tests['direct']['amp_icu_version']             = [
-			'label' => esc_html__( 'ICU version', 'amp' ),
-			'test'  => [ $this, 'icu_version' ],
-		];
-		$tests['direct']['amp_css_transient_caching']   = [
+
+		if ( function_exists( 'idn_to_utf8' ) ) {
```

Comment:
I suppose this makes sense because if `idn_to_utf8()` is not defined then the `intl` extension won't be installed, and for sure the `INTL_ICU_VERSION` constant won't be defined. 

However, if the site doesn't have an IDN, then they don't need `idn_to_utf8()` either and shouldn't be warned about it.

Therefore, I recommend not using `idn_to_utf8()` to determine whether they need the `INTL_ICU_VERSION` constant. Instead, I suggest adding a new method like: `is_intl_extension_needed()`. Then use this method in the `SiteHealth::add_extensions` method to determine whether to add the `intl` array key. Then also use this new `is_intl_extension_needed()` method here for whether to warn about the 'INTL_ICU_VERSION` constant being absent. 

So then can `is_intl_extension_needed()` checking for the the need of `intl` be done without `idn_to_utf8()`? Can it not be done simply by checking if the domain (or subdomain too?) has `xn--`?

Reply:
> Can it not be done simply by checking if the domain (or subdomain too?) has xn--?

That would not work If I we're to set the site URL to `https://⚡.com`, for example. The domain would have to be converted to IDNA ASCII first before being compared.

Reply:
I suppose we could check if the domain begins with `xn--` or if it contains unicode, then the `intl` extension is needed.

Reply:
It looks like the `xn--` prefix can be part of a domain or subdomain: https://serverfault.com/a/142805/127453

So we'd need to split the domain by `.`, remove the TLD (e.g. `com`), and then check if any of the segments begin with `xn--`.

In regards to determining whether we need `idn_to_utf8()` in the first place, for my own edification since I'm not an expert on IDN:

* `idn_to_utf8('xn--57h.com', IDNA_DEFAULT, INTL_IDNA_VARIANT_UTS46) === '⚡.com`'
* `idn_to_ascii('⚡.com', IDNA_DEFAULT, INTL_IDNA_VARIANT_UTS46) === 'xn--57h.com`'

According to the [AMP docs](https://amp.dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cache-urls/):

> When possible, the Google AMP Cache will create a subdomain for each AMP document's domain by first converting it from IDN (punycode) to UTF-8. The caches replaces every - (dash) with -- (2 dashes) and replace every . (dot) with - (dash). For example, pub.com will map to pub-com.cdn.ampproject.org.

This being the case, if the domain is domain is already in UTF-8, then we don't have to do anything to transform it. So that means we are only concerned with hostnames that have a the IDN prefix `xn--` in the domain or a subdomain. In other words, IDN is required if and only if this is true:

```php
preg_match( '/(^|\.)xn--/', $host )
```

Reply:
See https://github.com/ampproject/amp-wp/pull/4698/commits/b97582b56d53792c22caee39a4de3bf31dca2587 for changes discussed above.